java/arcs/core/storage/StorageProxy.kt

```diff
@@ -189,53 +232,69 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
     suspend fun onMessage(message: ProxyMessage<Data, Op, T>) {
         log.debug { "onMessage: $message" }
         when (message) {
-            is ProxyMessage.ModelUpdate -> {
-                val (futuresToResolve, value) = syncMutex.withLock {
-                    crdt.merge(message.model)
-                    isSynchronized = true
-                    val toResolve = waitingSyncs.toList() // list guaranteed to be copy
-                    waitingSyncs.clear()
-                    toResolve to crdt.consumerView
+            is ProxyMessage.ModelUpdate -> processModelUpdate(message.model)
+            is ProxyMessage.Operations -> processModelOps(message.operations)
+            is ProxyMessage.SyncRequest -> {
+                // Storage wants our latest state.
+                val modelUpdate = syncMutex.withLock {
+                    ProxyMessage.ModelUpdate<Data, Op, T>(crdt.data, null)
                 }
+                store.onProxyMessage(modelUpdate)
+            }
+        }
+    }
 
-                futuresToResolve.forEach { it.complete(value) }
+    private suspend fun processModelUpdate(model: Data) {
+        val (futuresToResolve, particleView, notifyFn) = syncMutex.withLock {
+            // TODO: send the returned merge changes to notifyUpdate()
```

Comment:
As part of this PR? Or a `b/####` as reference?

Reply:
I thought we shouldn't be putting anything that references google internals in external code?

Reply:
I have seen `b/` and `cl/` links mentioned in code and Github comments, so I presumed this meant we consider them low-signal enough to be OK. I would love for someone more experienced to clarify this for me once and for all.

Reply:
I can only see one b/ reference (in PeriodicCleanupTask.kt) and no cl/ in the code. They're pretty opaque though, so I can't see any harm in having them.